pkg/internal/kube/informer_provider.go

```diff
@@ -23,6 +23,16 @@ const (
 	kubeConfigEnvVariable = "KUBECONFIG"
 )
 
+// Some cloud providers store the cluster name as a Node label.
+// This greatly facilitates the retrieval of the cluster name, as we
+// don't need to rely on provider-specific APIs.
+// TODO: update with labels from other providers, or newer labels as long as specs are updated
+var clusterNameNodeLabels = []string{
+	"alpha.eksctl.io/cluster-name",
```

Comment:
Do you think this might work for AWS ECS on EC2? I guess in that case people will likely be using sidecars and it's less likely to be an issue. I'm not sure where you found these :), but there might be something for ECS too.

Reply:
I don't think it will work in ECS as this relies in the Kubernetes API. But actually this cluster name resolution is currently only performed on network metrics, inside Kubernetes clusters.

I think we should expand cluster name resolution to app metrics and non-kubernetes environments.